StarryExpanse/RivenGameState.cpp

```diff
@@ -1,16 +1,53 @@
 #include "RivenGameState.h"
+#include "SaveGame/RivenSaveGame.h"
+#include "Interfaces/RivenSaveGameAware.h"
 #include "Engine/Engine.h"
 #include "Kismet/GameplayStatics.h"
 #include "RivenGameInstance.h"
-#include "SaveGame/RivenSaveGame.h"
 #include "StarryExpanse.h"
+#include "Runtime/Core/Public/UObject/WeakObjectPtrTemplates.h"
+#include "Runtime/Core/Public/GenericPlatform/GenericPlatform.h"
 #include "UObject/ConstructorHelpers.h"
 
 ARivenGameState::ARivenGameState() : Super() {}
 
 void ARivenGameState::OnConstruction(const FTransform &Transform) {
   auto initialSavegame = NewObject<URivenSaveGame>();
+  initialSavegame->SetSubscriber(this);
   this->Instantaneous_SaveGame = initialSavegame;
   auto gameInstance = GetWorld()->GetGameInstance<URivenGameInstance>();
   gameInstance->Last_Savable_SaveGame = initialSavegame;
 }
+
+void ARivenGameState::SubscribeActorToSavegame(AActor* actor) {
+  if (actor != nullptr) {
```

Comment:
Can actor ever legitimately be null here, or is it always a programmers error of that happens? If the latter then I'd suggest using `check(actor)` instead.

Reply:
It should not be null, but that is actually an easy thing to accidentally do in Blueprints. If we use `check`, it'll crash, think we should?

Reply:
I'm really worried about silent failures. At a minimum we should emit an error log message. I think that once we get tests up and running we should consider adding checks to make it more likely that we'll catch the invalid actor at the time of Blueprint change instead of much later on.

Reply:
That's a good point. Long term maybe we should have something that crashes the game in development mode but silently continues for production builds.